docs/source/migration.mdx

```diff
@@ -580,580 +747,456 @@ method.
+Apollo Server 4 refactors the `GraphQLRequest` object. (This object is available to plugins as `requestContext.request`, and is the argument to `server.executeOperation`.) Specifically, its `http` field is now a new `HTTPGraphQLRequest` type instead of a type based on the Fetch API `Request` object. This object does not contain the URL path, and its `headers` field is a `Map` (with lower-case keys) rather than a Fetch API `Headers` object.
 
-```ts
-import express from 'express';
-import http from 'http';
-import cors from 'cors';
-import { json } from 'body-parser';
-import { ApolloServer, ApolloServerPluginDrainHttpServer } from '@apollo/server';
-import { expressMiddleware } from '@apollo/server/express4';
+### `GraphQLResponse`
 
-import { typeDefs, resolvers } from ('./schema');
+Apollo Server 4 refactors the [`GraphQLResponse` object](https://github.com/apollographql/apollo-server/blob/version-4/packages/server/src/externalTypes/graphql.ts#L25). (This object is available to plugins as `requestContext.response`, and is returned by `server.executeOperation`.) The `data`, `errors`, and `extensions` fields are now nested within an object returned by the `result` field:
 
-const server = new ApolloServer({
-  typeDefs,
-  resolvers,
-});
-
-const app = express();
-const httpServer = http.createServer(app);
+```ts disableCopy
+export interface GraphQLResponse {
+  // The below result field contains an object with the
+  //  data, errors, and extensions fields
+  result: FormattedExecutionResult;
+  http: HTTPGraphQLHead;
+}
+```
 
-// Add our ApolloServerPluginDrainHttpServer plug in *before* starting the server
-server.addPlugin(ApolloServerPluginDrainHttpServer({ httpServer }));
+Additionally, the  `data` and `extensions` fields are both type `Record<string, unknown>`, rather than `Record<string, any>`.
 
-await server.start();
-// Set up express middleware with body-parsing and cors
-app.use(cors(), json(), expressMiddleware(server));
-await new Promise<void>((resolve) => httpServer.listen({ port: 4000 }, resolve));
+`http.headers` is now a `Map` (with lower-case keys) rather than a Fetch API `Headers` object.
 
-console.log(`🚀 Server ready at http://localhost:4000`);
-```
+Note: we plan to implement experimental support for incremental delivery (`@defer`/`@stream`) before the v4.0.0 release, and we expect that this will change the structure of `GraphQLResponse` further.
 
-```js
-import express from 'express';
-import http from 'http';
-import cors from 'cors';
-import { json } from 'body-parser';
-import { ApolloServer, ApolloServerPluginDrainHttpServer } from '@apollo/server';
-import { expressMiddleware } from '@apollo/server/express4';
 
-import { typeDefs, resolvers } from ('./schema');
 
-const server = new ApolloServer({
-  typeDefs,
-  resolvers,
-});
+### Changes to plugin semantics
 
-const app = express();
-const httpServer = http.createServer(app);
+`requestDidStart` hooks are called in parallel rather than in series.
```

Comment:
I like the direction of the change but the point is that you can register multiple requestDidStart hooks, and this wording seems to detract from that.

Reply:
Gotcha! I'd like to give the sentence a little context, how does this feel: 

```suggestion
In Apollo Server 4, `requestDidStart` hooks are called in parallel rather than in series.
```